### PR TITLE
Fix ssh keys nohome

### PR DESCRIPTION
**Pull request by jreidinger**
## Problem

When in autoyast profile home path is not specied, then ssh keys are not written.

- https://bugzilla.suse.com/show_bug.cgi?id=1201185

## Solution

Read system users again before writting ssh keys to know its home paths.

## Testing

- *Unit tests adapted*
- *Tested manually*

### CODE REVIEW

File: src/lib/y2users/linux/users_writer.rb [hunk 1]
```diff
@@ -293,9 +319,7 @@ def adapt_home_ownership(user)
       # @param user [User]
       # @param previous_keys [Array<String>] previous auth keys for given user, if any
       # @return [Boolean] true on success
-      def write_auth_keys(user, previous_keys = [])
-        return true unless exist_user_home?(user)
```
Comment (joseivanlopez) [inline src/lib/y2users/linux/users_writer.rb:297]: Why isn't home checked? Would it fail if the user was created without home?

Comment (joseivanlopez) [reply src/lib/y2users/linux/users_writer.rb:297]: I see, an issue is generated when `Yast::Users::SSHAuthorizedKeyring::PathError` is raised.

File: test/lib/y2users/linux/users_writer_test.rb [hunk 2]
```diff
@@ -53,6 +53,12 @@
 
   let(:commit_config) { Y2Users::CommitConfig.new }
 
+  let(:system_config) { initial_config }
+
+  before do
+    allow(Y2Users::Linux::Reader).to receive(:new).and_return(double(read: system_config))
```
Comment (joseivanlopez) [inline test/lib/y2users/linux/users_writer_test.rb:59]: NP: I would use `instance_double`

File: test/lib/y2users/linux/users_writer_test.rb [hunk 3]
```diff
@@ -331,56 +337,44 @@ def issues(messages)
...
 
         context "and the authorized keys has changed" do
+          let(:system_user) { target_user.copy }
+
           before do
             target_user.authorized_keys = ["new-key"]
+            # cannot overwrite here system_config as we need to have there recent
+            # target user which is modified in `before` code
+            system_config = Y2Users::Config.new.tap { |c| c.attach([system_user]) }
+            allow(Y2Users::Linux::Reader).to receive(:new).and_return(double(read: system_config))
```
Comment (joseivanlopez) [inline test/lib/y2users/linux/users_writer_test.rb:347]: And here too.

File: test/lib/y2users/linux/users_writer_test.rb [hunk 4]
```diff
@@ -331,56 +337,44 @@ def issues(messages)
         end
 
         context "and the authorized keys has changed" do
+          let(:system_user) { target_user.copy }
+
           before do
             target_user.authorized_keys = ["new-key"]
+            # cannot overwrite here system_config as we need to have there recent
+            # target user which is modified in `before` code
+            system_config = Y2Users::Config.new.tap { |c| c.attach([system_user]) }
```
Comment (joseivanlopez) [inline test/lib/y2users/linux/users_writer_test.rb:346]: NP: you don't need to pass a list to `#attach`.

File: src/lib/y2users/linux/users_writer.rb (hunk 1 above)
Comment (jreidinger) [reply src/lib/y2users/linux/users_writer.rb:297]: yes and I also think that on system it is not possible to have user without home. at least man page does not mention it and also my own passwd does not have single entry without home. Even one without real home has something like `/bin` or `/sbin`

Comment (joseivanlopez) [reply src/lib/y2users/linux/users_writer.rb:297]: You are right. Checking the code, I see that the home can be created as empty (without skel) but the directory is always created.

**Review by joseivanlopez — APPROVED** [review]
LGTM

**Pull request merged**